push-build.sh

```diff
@@ -106,22 +108,22 @@ RELEASE_BUCKET=${FLAGS_bucket:-"kubernetes-release-dev"}
 # This will canonicalize the path
 KUBE_ROOT=$(pwd -P)
 
-KUBECTL_OUTPUT=$(cluster/kubectl.sh version --client 2>&1 || true)
-if [[ "$KUBECTL_OUTPUT" =~ GitVersion:\"(${VER_REGEX[release]}(\.${VER_REGEX[build]})?(-dirty)?)\", ]]; then
+gitlib::get_version_vars
```

Comment:
Why not just rely on incoming LATEST as set by the calling scope?  How/why is this used differently here?

Reply:
IIUC, LATEST is not set by calling scope earlier too. it is always set in this code block by getting GitVersion from `cluster/kubectl.sh version --client`.
Using kubectl to get version is not generic and moreover kubectl does not exist in federation repo/release. So instead i replaced it with git by calling function `get_version_vars` which is copied from `kubernetes/hack/lib/version.sh`.

Reply:
`push-build.sh` can take `--version` in from the command-line too, though if you really need the specific functionality of `get_version_vars` then this becomes problematic.  I suppose we're bumping up against the issue of splitting up the org repos and trying to share code.
@ixdy @BenTheElder with the way test-infra is using `push-build.sh` and incoming `"LATEST"/--version` values, what do you think makes sense here?

We probably need some version parsing api layer easily shared between the repos.  I'm loathe to start making full on copies of functions all over the place.

Reply:
@david-mcmahon, i scanned the `push-build.sh` file multiple times and i could not find the `--version` flag. IIUC, `LATEST` variable was fetched from output of `kubectl version --client` from within `push-build.sh` file.

@ixdy, @BenTheElder request to PTAL. we are on a tight schedule to move over to out-of-core  federation.

Reply:
I don't think test-infra is setting a `--version` flag for push-build.sh, usages are here:
https://github.com/kubernetes/test-infra/blob/6246863dfb075830d67c99f48559aafcad3ddf2b/kubetest/stage.go#L67
https://github.com/kubernetes/test-infra/blob/0714e1db9658ab70b8a1a3c42aeaaf88c43159a4/scenarios/kubernetes_build.py#L49

For test-infra I would prefer to move things to bazel rules (`bazel run //:push-build gcs://base-bucket/path`) which cleanly solves logic sharing I think.

For this script though an option I explored before (with a pointer from @ixdy) was to use the version file we build in the tarball instead of parsing kubectl. We do something like this [here](https://github.com/kubernetes/test-infra/blob/75b80778e177dd1fa8dcc198b3448167620b2952/scenarios/kubernetes_bazel.py#L59) for the bazel build job, but any release tarball should contain a version file.

Reply:
Thanks for the suggestion @BenTheElder. Using the version file will work for federation release. I will update the pr accordingly.